File: backend/routes/geofence.js
const express = require('express');
const router = express.Router();
const authMiddleware = require('../middleware/auth');

/**
 * GET /api/geofences
 * Get all geofence definitions
 */
router.get('/geofences', authMiddleware, async (req, res) => {
  try {
    // In a real implementation, you would fetch from database
    // For now, returning a hardcoded office location
    const geofences = [
      {
        id: '1',
        name: 'Main Office',
        latitude: 37.7749,
        longitude: -122.4194,
        radiusMeters: 100,
        createdAt: new Date('2025-01-01'),
        updatedAt: new Date(),
      },
    ];

    res.json({
      success: true,
      geofences,
    });
  } catch (error) {
    res.status(500).json({
      success: false,
      message: 'Failed to fetch geofences',
    });
  }
});

module.exports = router;

File: attendance_app/lib/src/domain/models/geofence.dart
import 'package:freezed_annotation/freezed_annotation.dart';

part 'geofence.freezed.dart';
part 'geofence.g.dart';

/// Represents a geofence zone (e.g., office location)
@freezed
class Geofence with _$Geofence {
  const factory Geofence({
    required String id,
    required String name,
    required double latitude,
    required double longitude,
    /// Radius in meters
    required double radiusMeters,
    required DateTime createdAt,
    DateTime? updatedAt,
  }) = _Geofence;

  factory Geofence.fromJson(Map<String, dynamic> json) =>
      _$GeofenceFromJson(json);
}
